src/history/hs.c

```diff
@@ -1341,6 +1341,21 @@ __wt_hs_find_upd(WT_SESSION_IMPL *session, WT_ITEM *key, const char *value_forma
                 break;
             }
 
+            /*
```

Comment:
I'm a bit concerned that we still seem to need this even with the other changes. I'll investigate this more tomorrow.

Reply:
Oh, actually this makes sense. When I removed this, I added:

```
+            /* History store records should never be newer than the on-disk value. */
+            WT_ASSERT(session,
+              on_disk_tw->start_ts > hs_start_ts_tmp ||
+                (on_disk_tw->start_ts == hs_start_ts_tmp &&
+                  on_disk_tw->start_txn > hs_cbt->upd_value->tw.start_txn));
```

Anytime a failed eviction with multiple updates newer than the on-disk value, this can happen and it isn't harmful.